### PR TITLE
Fix filter scrollarea not adjusting to parent height

### LINKED ISSUE #2174: Filter section scroll area not adapting to window height
### Describe the issue:

- [x] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:

When viewing the filter options in the lower right side of the screen, the scrollbar is not adapting when the height of the screen area available to it is changed.

### Operating system:

Mac OS X 10.14.5

### Subsurface version:

Latest master branch (4.8.4-682-g5886550434e2)

### Steps to reproduce:

1) Open filter window with <Command-F>
2) Adjust height of Filter Selection region

### Current behavior:

Scroll region does not change height when the area height is changed.

### Expected behavior:

Scroll region should use the full window area available to it.

### Additional information:

This was also reported in a comment on #2152 after that issue was closed.
Screen Shot 2019-07-11 at 7 42 30 AM

### PR DESCRIPTION
**Pull request by djunkins**
Signed-off-by: Doug Junkins <junkins@foghead.com>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Changed the Qt Layout for FilterWidget2 to "Lay Out in a Grid" to
allow the scrollarea to adjust when the height of FilterWidget2
changes. 

### Related issues:

Fixes #2174

### CODE REVIEW

**Comment by djunkins** [comment]
I'm not sure why the Travis check failed. The logs seem to be truncated. Could it have just timed out?

File: desktop-widgets/filterwidget2.ui [hunk 1]
```diff
@@ -13,461 +13,457 @@
...
+     <property name="widgetResizable">
+      <bool>true</bool>
+     </property>
+     <widget class="QWidget" name="scrollAreaWidgetContents">
+      <property name="geometry">
+       <rect>
+        <x>0</x>
+        <y>0</y>
+        <width>486</width>
+        <height>487</height>
```
Comment (dirkhh) [inline desktop-widgets/filterwidget2.ui:34]: these values always bug me... I know they are auto-generated, but they often cause us to suddenly have much bigger minimum height of the widget

**Review by dirkhh — COMMENTED** [review]
I need to build this locally and make sure it doesn't undo the fix to the minimum height... that's something we need to find an automated way to test - not the faintest idea how that would work.
But we have a tendency to allow the app to require more and more minimum screen space - and people traveling with older laptops get screwed.

**Comment by dirkhh** [comment]
(editorial comment... for changes to `.ui` files the 'ignore whitespace' mode on GitHub is heaven sent...)

File: desktop-widgets/filterwidget2.ui (hunk 1 above)
Comment (dirkhh) [reply desktop-widgets/filterwidget2.ui:34]: tested and it doesn't seem to affect minimum height...

**Review by dirkhh — APPROVED** [review]
let's merge this

**Pull request merged**